provider/server/machine_delete_test.go

```diff
@@ -14,11 +15,111 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-// TODO: This test will require update after merge of PR: #101
 var _ = Describe("DeleteMachine", func() {
 
-	It("should delete a machine", func(ctx SpecContext) {
-		By("creating a machine")
+	It("should delete a machine with graceful shutdown", func(ctx SpecContext) {
+		By("creating a machine using squashfs os image which boots properly")
+		createResp, err := machineClient.CreateMachine(ctx, &iri.CreateMachineRequest{
+			Machine: &iri.Machine{
+				Metadata: &irimeta.ObjectMetadata{
+					Labels: map[string]string{
+						"foo": "bar",
+					},
+				},
+				Spec: &iri.MachineSpec{
+					Power: iri.Power_POWER_ON,
+					Image: &iri.ImageSpec{
+						Image: squashfsOSImage,
+					},
+					Class: machineClassx3xlarge,
+					Volumes: []*iri.Volume{
+						{
+							Name: "disk-1",
+							EmptyDisk: &iri.EmptyDisk{
+								SizeBytes: emptyDiskSize,
+							},
+							Device: "oda",
+						},
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(createResp).NotTo(BeNil())
+
+		By("ensuring domain and domain XML is created for machine")
+		var domain libvirt.Domain
+		Eventually(func() error {
+			domain, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+			return err
+		}).WithTimeout(5 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
+		domainXMLData, err := libvirtConn.DomainGetXMLDesc(domain, 0)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(domainXMLData).NotTo(BeEmpty())
+
+		By("ensuring domain for machine is in running state")
+		Eventually(func() libvirt.DomainState {
+			domainState, _, err := libvirtConn.DomainGetState(domain, 0)
+			Expect(err).NotTo(HaveOccurred())
+			return libvirt.DomainState(domainState)
+		}).Should(Equal(libvirt.DomainRunning))
+
+		By("ensuring machine is in running state")
+		Eventually(func() iri.MachineState {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).NotTo(BeEmpty())
+			return listResp.Machines[0].Status.State
+		}).Should(Equal(iri.MachineState_MACHINE_RUNNING))
+
+		//allow some time for the vm to boot properly
+		time.Sleep(30 * time.Second)
```

Comment:
Sleep is good enough for now but is not the best solution here. Alternatively we could think of using `Eventually` and try to wait until the SSH port of the machine is open. We should be able to use a local network device as a `NetworkInterface` here.

Reply:
@Rohit-0505 could you pls create issue for it. we will have the same problem in another tests too. So we need one proper solution for boot waiting.

Reply:
yeah I will create one